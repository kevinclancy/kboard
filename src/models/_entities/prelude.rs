//! `SeaORM` Entity, @generated by sea-orm-codegen 1.1.13

pub use super::boards::Entity as Boards;
pub use super::replies::Entity as Replies;
pub use super::reply_statuses::Entity as ReplyStatuses;
pub use super::threads::Entity as Threads;
pub use super::users::Entity as Users;
