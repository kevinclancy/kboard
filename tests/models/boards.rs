use kboard::{app::App, models};
use loco_rs::testing::prelude::*;
use serial_test::serial;
use insta::assert_debug_snapshot;

macro_rules! configure_insta {
    ($($expr:expr),*) => {
        let mut settings = insta::Settings::clone_current();
        settings.set_prepend_module_to_snapshot(false);
        settings.set_snapshot_suffix("boards");
        let _guard = settings.bind_to_scope();
    };
}

#[tokio::test]
#[serial]
async fn test_findall() {
    configure_insta!();

    let boot = boot_test::<App>().await.unwrap();
    seed::<App>(&boot.app_context).await.unwrap();

    let boards = models::boards::Model::find_all(&boot.app_context.db).await;

    assert_debug_snapshot!(boards);
}
