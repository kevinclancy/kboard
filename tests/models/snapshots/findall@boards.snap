---
source: tests/models/boards.rs
expression: boards
---
Ok(
    [
        Model {
            created_at: 2023-11-12T12:34:56.789+00:00,
            updated_at: 2023-11-12T12:34:56.789+00:00,
            id: 1,
            title: "Broken Jaw Stories",
            description: "Tell about the time you broke your jaw.",
            num_threads: 4,
        },
    ],
)
